sdk/helper/awsutil/region.go

```diff
@@ -14,7 +14,8 @@ import (
 // is a widely used region, and is the most common one for some services like STS.
 const DefaultRegion = "us-east-1"
 
-var ec2MetadataBaseURL = "http://169.254.169.254"
+// This is nil by default, but is exposed in case it needs to be changed for tests.
+var ec2Endpoint *string = nil
```

Comment:
very minor nit: the `= nil` here is redundant, `var thing *type` will result in a variable with nil value for `type`